### ISSUE TITLE
`total=False` lost on `DeviceInfo`

### ISSUE THREAD

**Issue by scop** [KapJI/homeassistant-stubs#291]
Perhaps the same issue as #290, but thought I'd report it anyway:

https://github.com/home-assistant/core/blob/d0a9d658754b5c106571a92a79afdf4c7c8488a7/homeassistant/helpers/entity.py#L163 vs
https://github.com/KapJI/homeassistant-stubs/blob/98df454159346546581bd0cd540c1945870d46cb/homeassistant-stubs/helpers/entity.pyi#L36
-- `total=False` got lost.

I suppose this is why for something like https://github.com/scop/home-assistant-jatekukko/blob/66ceb35160f20026d3f638e7696513a585a5436f/custom_components/jatekukko/coordinator.py#L30, mypy gives:
```
custom_components/jatekukko/coordinator.py:30: error: Missing keys ("connections", "default_name", "manufacturer", "model", "suggested_area", "sw_version", "hw_version", "via_device") for TypedDict "DeviceInfo"  [typeddict-item]
```

This is with homeassistant-stubs 2023.2.4.

**Comment by bramstroker** [comment]
I have this same issue.
For example on `NumberSelectorConfig(TypedDict)`, total arg is not in the stub.

Also got numerous errors on my custom component.
So ended up disabling that error for now:

`disable_error_code = "typeddict-item`

But this is not really a solution and would be great if the stub reflects the signature of the real dict.

**Comment by bramstroker** [comment]
Seems like a bug in mypy stubgen tool to me.
I have created a bug report https://github.com/python/mypy/issues/15195

**Comment by bramstroker** [comment]
This has already been fixed in stubgen, however not released yet in mypy 1.3.0
I have checked with master branch of mypy and now it's keeping the `Total` attribute correctly. 
So we still need to wait a bit for next stable release.

**Comment by KapJI** [comment]
Thanks for chasing this with mypy team! Hopefully the fix will be included in the next release.

**Comment by KapJI** [comment]
Fix should be released in mypy 1.4.

**Comment by bramstroker** [comment]
Think this issue can be closed. Everything is working correctly now with MyPy 1.4.0 and homeassistant-stubs 2023.6.3
